cmd/manager/main.go

```diff
@@ -130,22 +145,33 @@ func main() {
 	dependentSelector := k8slabels.NewSelector().Add(*dependentRequirement)
 
 	setupLog.Info("set up manager")
+	cacheOptions := crcache.Options{
+		ByObject: map[client.Object]crcache.ByObject{
+			&ocv1alpha1.ClusterExtension{}: {Label: k8slabels.Everything()},
+			&catalogd.ClusterCatalog{}:     {Label: k8slabels.Everything()},
+		},
+		DefaultNamespaces: map[string]crcache.Config{
+			systemNamespace: {LabelSelector: k8slabels.Everything()},
+		},
+		DefaultLabelSelector: dependentSelector,
+	}
+	if globalPullSecretKey != nil {
+		cacheOptions.ByObject[&corev1.Secret{}] = crcache.ByObject{
+			Field: fields.SelectorFromSet(map[string]string{
+				"metadata.name": globalPullSecretKey.Name,
+			}),
+		}
+		cacheOptions.DefaultNamespaces[globalPullSecretKey.Namespace] = crcache.Config{
```

Comment:
So the answer for these can be anchored around the RBAC. 

- The Role and the RoleBinding for our SA must exist before we can set up a watch from our controller. 
- If the namespace does not exist, the Role/RoleBinding cannot exist. So the controller pod will keep crashing 
```
E1004 12:06:32.363508       1 reflector.go:158] "Unhandled Error" err="pkg/cache/internal/informers.go:106: Failed to watch *v1.Secret: failed to list *v1.Secret: secrets \"pull-secret\" is forbidden: User \"system:serviceaccount:olmv1-system:operator-controller-controller-manager\" cannot list resource \"secrets\" in API group \"\" in the namespace \"test-namespace\"" logger="UnhandledError"
```

The last two questions were interesting finds for me. Looks like if the namespace exists when the controller starts but then is deleted, I don't see any error immediately. Nor for the ~5 mins that I kept the `logs -f` up. 
When the namespace gets created, I still don't see anything in the logs. 

TLDR of course is that you must have the proper Role/Rolebinding (and as a result the namespace) before you start our controller with the flag to watch for the specific secret in that namespace. Otherwise the controller will fail to start.

Reply:
> If the namespace does not exist, the Role/RoleBinding cannot exist.

Yeah, in my scenarios let's assume that there is a ClusterRole and ClusterRoleBinding, so that we can assume we have permission everywhere all the time. (That obviously wouldn't be what we'd suggest, but its useful to isolate the behavior of the caching logic and informers)

I don't think we need to actually _solve_ any weirdness that comes up with my scenarios. I just think we need to understand it and write it down in docs and in the help text of the flag we add.

For example, I could see us saying:
```
--global-pull-secret    A secret reference with the format <namespace>/<name>. The namespace is expected to exist for the life of the process. Creation or deletion of the namespace after the controller starts is likely to ... <whatever we see happens when we test this>"
```

Reply:
Agree with the approach of saying that "this and that is a prerequisite" and going simple. We. can come up with something more sophisticated later, if we see the need for it.

Reply:
That is a LONG message for the flag though. Probably neat-er to document this in the doc we write for this.

Reply:
I think it is fine. Kubectl has longer messages, for example. And this one is something which gets configured once and probably rarely modified.

But adding it into docs in addition to the flag description sounds good to me to.